### PR TITLE
Friesian joingroupby, target encode

### PR DESCRIPTION
**Pull request by yaxinchen666**
Currently no cross validation in target encode.

### CODE REVIEW

**Comment by jason-dai** [comment]
Please split the PR to smaller ones (i.e., join_groupby, difference_lag, and target_ecoding)

File: pyzoo/zoo/friesian/feature/utils.py [hunk 1]
```diff
@@ -62,3 +62,37 @@ def check_col_exists(df, columns):
...
+def gen_cols_name(columns, name_sep="_"):
+    if isinstance(columns, str):
+        return columns
+    elif isinstance(columns, list):
+        return name_sep.join(columns)
+    else:
+        raise ValueError("item should be either str or list of str")
+
+
+def str_to_list(arg_name, arg):
```
Comment (jason-dai) [inline pyzoo/zoo/friesian/feature/utils.py:94]: change to `def str_to_list(arg, arg_name)`

Comment (yaxinchen666) [reply pyzoo/zoo/friesian/feature/utils.py:94]: Got it. I will modify it in new PRs.

**Comment by hkvision** [comment]
Split into separate ones. Closing this one.

**Pull request closed without merging**